src/kahypar.jl

```diff
@@ -148,11 +148,21 @@ end
 
 function recursive_bipartite_optimize(bipartiter, code::EinCode, size_dict)
     ixs, iy = getixsv(code), getiyv(code)
-    ixv = [ixs..., iy]
+    if isempty(iy)
+        ixv = [ixs..., iy]
+    else
+        ixv = [ixs..., iy, empty(iy)]
+    end
+    
     adj, edges = adjacency_matrix(ixv)
-    vertices=collect(1:length(ixs))
+    vertices=collect(1:length(ixv) - 1)
```

Comment:
I am not sure if it is still correct for empty `iy`.

Reply:
Currently for empty `iy` the results correct, since `ixv = [ixs..., iy]`, then `length(ixv) - 1 == length(ixs)` and the process is exactly the same to the previous version.